packages/inquirer/lib/inquirer.js

```diff
@@ -4,37 +4,45 @@
  * A collection of common interactive command line user interfaces.
  */
 
-const inquirer = module.exports;
+import { default as List } from './prompts/list';
+import { default as Input } from './prompts/input';
+import { default as Number } from './prompts/number';
+import { default as Confirm } from './prompts/confirm';
+import { default as RawList } from './prompts/rawlist';
+import { default as Expand } from './prompts/expand';
+import { default as Checkbox } from './prompts/checkbox';
+import { default as Password } from './prompts/password';
+import { default as Editor } from './prompts/editor';
+
+import { default as BottomBar } from './ui/bottom-bar';
+import { default as Prompt } from './ui/prompt';
+
+export { default as Separator } from './objects/separator';
 
 /**
  * Client interfaces
  */
-
-inquirer.prompts = {};
-
-inquirer.Separator = require('./objects/separator');
-
-inquirer.ui = {
-  BottomBar: require('./ui/bottom-bar'),
-  Prompt: require('./ui/prompt'),
+export const ui = {
+  BottomBar,
+  Prompt,
 };
 
 /**
  * Create a new self-contained prompt module.
  */
-inquirer.createPromptModule = function (opt) {
+export function createPromptModule(opt) {
```

Comment:
This changes the public interface slightly since it removes members of the exported object.

I find this syntax better, but we should keep both for easier backward compatibility. e.g

```js
function createPromptModule(opt) { ... }

inquirer.createPromptModule = createPromptModule;

export { createPromptModule }
```

(and etc for the other functions exposed here)

Reply:
```ReferenceError: module is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and '/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
    at file:///Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/lib/inquirer.js:22:18
    at ModuleJob.run (node:internal/modules/esm/module_job:198:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:409:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:435:15)
    at async formattedImport (/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/node_modules/mocha/lib/nodejs/esm-utils.js:7:14)
    at async exports.requireOrImport (/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/node_modules/mocha/lib/nodejs/esm-utils.js:48:32)
    at async exports.loadFilesAsync (/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/node_modules/mocha/lib/nodejs/esm-utils.js:103:20)
    at async singleRun (/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/node_modules/mocha/lib/cli/run-helpers.js:125:3)
    at async exports.handler (/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/node_modules/mocha/lib/cli/run.js:374:5)```

Reply:
Since we converted to an ESModule, module does not exist in ES Module syntax

Reply:
Right, so you'd change things a bit too:

```js
export function createPromptModule(opt) { ... }

const inquirer = {
    createPromptModule,
    ...
};

export default inquirer;
```

Reply:
I modified the code directly to keep backward compat export. Should work fine now